designer/client/src/components/graph/node-modal/editors/expression/Table/TableEditor.tsx

```diff
@@ -97,7 +97,7 @@ export const Table = ({ expressionObj, onValueChange, className, fieldErrors }:
         }
     }, [expressionObj.expression, onValueChange, rawExpression]);
 
-    const { defaultTypeOption, orderedTypeOptions } = useTypeOptions<SupportedType>();
+    const { defaultTypeOption, orderedTypeOptions } = useTypeOptions<SupportedType>(false);
```

Comment:
Why do we need this change?, false is not a valid Supported type

Reply:
See: https://github.com/TouK/nussknacker/pull/7167#discussion_r1850207954

Reply:
It is certainly possible but I do not want to make this refactor. I tried to do it for `TableEditor` and spend on it some time but there is some internal dependency on `.refClazzName` inside it...

Reply:
So long story short is that `TableEditor` uses it _somehow_ and `FragmentInputDefinition` does not use it anymore and relies on simple class names which are resolved on BE.

Reply:
Ok, thanks for the explanation, two more questions 
1.  Based on [types](https://github.com/TouK/nussknacker/blob/49dc2be6f59f2f74fa2991ab0ac3023448c78a41/designer/client/src/types/definition.ts#L6) `.display` is required property, is it also true for `TableEditor`? If not, could you adjust this type?
2. If there is no `.display` in the case of `TableEditor`, could you simplify this logic and just make it `type.display  ||  type.refClazzName`?

Reply:
But there is `.display` in the `TableEditor` context. This works as follows: 

Request is performed to definition data. We obtain information containing `display` as well as `refClazzName`.  The problem is that we would like to use create `value` part of this hook-data depending on place where we call this hook (aka hook-use context). So the simplification `type.display  ||  type.refClazzName` will not working bcs this will be evaluated to `type.display` which should not be present in the context of `TableEditor`....

Reply:
Okay, I got it. Let's return to the implementation. To be honest, I prefer your first approach (separating it into two methods and adding meaningful names). Even if the code is duplicated, IMO readability is more important.
In the current approach, this boolean says nothing and I need to go into implementation details to check what's going on here

```
const { orderedTypeOptions, defaultTypeOption } = useTypeOptions<string>(true);
```

we could change it to an object

```
const { orderedTypeOptions, defaultTypeOption } = useTypeOptions<string>({ useDisplay:true });
```

but still, I don't know what happens when I set it to false, and what exactly happens when it is true.

Reply:
As we discussed in DM I would go with two different hooks.